### DISCUSSION TITLE
QskDrawer's problem

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#462] [Q&A] [answered]
When I set the DragMargin of the QskDrawer, no matter what value I set the DragMargin to, its area is not affected, but its DragMargin property will change. I can only control its size with setmargins
auto dwr=new QskDrawer(this);
         dwr->setEdge(Qt::RightEdge);
         auto btn=new ButtonPage(dwr);
         dwr->setDragMargin(1000);
         qDebug()<<dwr->dragMargin();
         dwr->open();

**Comment by uwerat** [comment, marked as answer]
The dragMargin is a parameter of the gesture - the drawer opens, once the movement exceeds that margin.

The size of the drawer is automatically set to its preferredSize. If you did not assign one manually ( QskControl::setPreferredSize ) the size is calculated from the sizeHints of its children.

**Comment by hestrro** [comment]
Just tried it，when the pop-up edge is up or down, only the height of the drawer can be changed. When the pop-up edge is left or right, only the width of the drawer can be changed.

**Comment by uwerat** [comment]
True - see qskAlignedToEdge at https://github.com/uwerat/qskinny/blob/master/src/controls/QskDrawer.cpp#L60

In case you don't want your content being aligned with the edges you can wrap it into a QskControl container item. The container will be aligned like above - however it has no scenegraph nodes. Then:

- use one of the layout classes as container
- enable QskControl::autoLayoutChildren )-
- calculate the geometries manually by overloading QskControl::updatelayout
